codelst:0.0.3

jneug opened the pull request:
<!--
Thanks for submitting a package! Please read and follow the submission guidelines detailed in the repository's README and check the boxes below. If you want to make a PR for something other than a package submission, just delete all this and make a plain PR.
-->

I am submitting
- [x] a new package
- [ ] an update for a package

I have read and followed the submission guidelines and, in particular, I
- [x] selected a name in conformance with the guidelines
- [x] added a `typst.toml` file with all required keys
- [x] added a `README.md` with documentation for my package
- [x] have chosen a license and added a `LICENSE` file or linked one in my `README.md`
- [x] ensured that my submission does not infringe upon the rights of a third party
- [x] tested my package locally on my system and it worked
- [x] named this PR as `name:version` of the submitted package
- [x] agree that my package will not be removed without good reason

<!--
Please add a brief description of your package below and explain why you think it is useful to others.
-->

Description: **codelst** adds useful features for rendering source code. It can add line numbers, load and show source code from files, highlight lines or add labels to specific lines to make references from other places.

As of now it lacks a comprehensive manual, but the examples should make the usage clear enough.

laurmaedje commented on packages/preview/codelst-0.0.1/README.md:
`````diff
@@ -0,0 +1,40 @@
+````typst
+#import "@preview/codelst:0.0.1": sourcecode
+
+#sourcecode[```typc
`````
```suggestion
#sourcecode[```typ
```

laurmaedje reviewed (commented):
The `sourcefile` function doesn't work unfortunately because the `read` call happens within the package. There is currently no way for a package to read files outside of itself. I also noticed that highlighting for some languages (Typst itself in particular) is suboptimal because each line is highlighted individually (might be a deliberate trade-off, but just wanted to mention it in case you are not aware).

jneug commented:
> The `sourcefile` function doesn't work unfortunately because the `read` call happens within the package. There is currently no way for a package to read files outside of itself. 

Thats unfortunate but understandable. I didn't test that properly. 

I will remove that function.

> I also noticed that highlighting for some languages (Typst itself in particular) is suboptimal because each line is highlighted individually (might be a deliberate trade-off, but just wanted to mention it in case you are not aware). 

I'm aware of that problem and hope to fix it in future releases. I experimented with different solutions, but they weren't as robust and broke for things like page breaks or long lines.

jneug commented:
I already did some minor fixes for 0.0.2 and added a manual. With the changes to `sourcefile` I bumped the version to 0.0.3 and updated the pull request accordingly.

`sourcefile` now takes the file content instead of the filename and expects the user to call `read` from the main document:
```js
#sourcefile(read("filename.typ"), lang:"typ")
```

laurmaedje commented:
Thank you!
